CalibPPS/ESProducers/test/scipt_test_many_writeTotemDAQMapping.py

```diff
@@ -0,0 +1,159 @@
+  "outputFileName": "all_timing",
+  "sampicSubDetId" : 5,
+  "configuration": cms.VPSet(
+    # 2017, before detector inserted in DAQ
+    cms.PSet(
+      validityRange = cms.EventRange("1:min - 310000:max"),
+      mappingFileNames = cms.vstring(),
+      maskFileNames = cms.vstring()
+    ),
+    # 2018
+    cms.PSet(
+      validityRange = cms.EventRange("310001:min - 339999:max"),
+      mappingFileNames = cms.vstring("CondFormats/PPSObjects/xml/mapping_totem_timing_2018.xml"),
+      maskFileNames = cms.vstring()
+    ),
+    # 2022
+    cms.PSet(
+      validityRange = cms.EventRange("340000:min - 999999999:max"),
+      mappingFileNames = cms.vstring("CondFormats/PPSObjects/xml/mapping_totem_timing_2022.xml"),
+      maskFileNames = cms.vstring()
+    )
+  )
+}
+
+#-----------------------------------------------------------------
+
+files_to_read = [diamonds]  
+from_db = False
+
+
+for file_content in files_to_read:
+    for file_info in file_content["configuration"]:
+        with open('CalibPPS/ESProducers/test/test_writeTotemDAQMapping.py', 'r+') as f:        
+            content = f.read()
+            content = re.sub(r'subSystemName =.*', f'subSystemName = "{file_content["subSystemName"]}"', content)
+            content = re.sub(r'process.CondDB.connect =.*', f'process.CondDB.connect = "{file_content["dbConnect"]}"', content)
+            content = re.sub(r'process.totemDAQMappingESSourceXML.sampicSubDetId =.*', f'process.totemDAQMappingESSourceXML.sampicSubDetId = cms.uint32({file_content["sampicSubDetId"]})', content)
+            content = re.sub(r'fileName =.*', f'fileName = cms.untracked.string("{file_content["outputFileName"] + "_db.txt" if from_db else file_content["outputFileName"]+"_xml.txt"}")', content)
+          
+            content = re.sub(r'min_iov =.*', f'min_iov = {file_info.validityRange.start()}', content)
+            content = re.sub(r'max_iov =.*', f'max_iov = {file_info.validityRange.end()}', content)
+            content = re.sub(r'mappingFileNames =.*', f'mappingFileNames = {file_info.mappingFileNames},', content)
+            content = re.sub(r'maskFileNames =.*', f'maskFileNames = {file_info.maskFileNames},', content)
+            
+            if from_db:
+                content = re.sub(r'process.es_prefer_totemTimingMapping =.*', 
+                                f'process.es_prefer_totemTimingMapping = cms.ESPrefer("PoolDBESSource", "", TotemReadoutRcd=cms.vstring(f"TotemDAQMapping/{file_content["subSystemName"]}"))', content)
+            else:                
+                content = re.sub(r'process.es_prefer_totemTimingMapping =.*', 
+                                f'process.es_prefer_totemTimingMapping = cms.ESPrefer("TotemDAQMappingESSourceXML","totemDAQMappingESSourceXML",TotemReadoutRcd=cms.vstring("TotemDAQMapping/{file_content["subSystemName"]}"))', content)            
+          
+            f.seek(0)
+            f.write(content)
+            f.truncate()
```

Comment:
Same issues as in [script-ctpps-write-many-XML-to-SQLite.py](https://github.com/CTPPS/cmssw/pull/79/files#diff-0e4fc4d568e2cae19bf49dd0c7cad074f6bc12fb0e8ed3cf69511660cd89711b):
* too long lines
* hard to understand: split to subfunctions or comment